test/functional/feature_block.py

```diff
@@ -1400,6 +1401,9 @@ def update_block(self, block_number, new_transactions):
         self.add_transactions_to_block(block, new_transactions)
         old_sha256 = block.sha256
         block.hashMerkleRoot = block.calc_merkle_root()
+        has_witness_tx = any(not tx.wit.is_null() for tx in block.vtx)
+        if has_witness_tx:
+            add_witness_commitment(block)
```

Comment:
Right now this code would not be executed at all, do you think it makes sense to add a new invalid tx with witness data to `invalid_txs.py`?

Similar to what you had [here](https://github.com/ajtowns/bitcoin/pull/13/files#diff-d390e9e12bfa34a462fdff9f1894d7dc3edb45c26cf55df486d92864a3052fafR139)
[this](https://github.com/bitcoin/bitcoin/commit/686ecf3a0c39652613635848b7efb93af2a74761) should execute `add_witness_commitment`